### PR TITLE
list all paginated hosted DNS zones for AWS Route53

### PR DESCRIPTION
**Pull request by yithian**

### CODE REVIEW

**Comment by yithian** [comment]
for OSD-7579

File: pkg/clients/aws/route53.go [hunk 1]
```diff
@@ -528,3 +528,26 @@ func getSTSCredentials(client *sts.STS, roleArn string, externalID string, roleS
 	}
 	return assumeRoleOutput, nil
 }
+
+func listAllHostedZones(r53 route53iface.Route53API, lhzi *route53.ListHostedZonesInput) ([]*route53.HostedZone, error) {
```
Comment (dustman9000) [inline pkg/clients/aws/route53.go:532]: To be consistent, can we add a comment to the purpose of the function? Although the name is pretty self-explanatory.

**Review by fahlmant — APPROVED** [review]
/lgtm
/hold

Looks good from my end. Unhold when ready.

File: pkg/clients/aws/route53.go (hunk 1 above)
Comment (yithian) [reply pkg/clients/aws/route53.go:535]: done!

**Review by fahlmant — APPROVED** [review]
/lgtm

**Comment by dustman9000** [comment]
Nice work!
/lgtm

**Pull request merged**